src/web_fetch.js

```diff
@@ -62,59 +65,73 @@ async function webFetch(url, options) {
         }
         throw err;
       }
-    };
-
-    const handleNfsFetchException = (error) => {
-      if (error.code !== errConst.ERR_FILE_NOT_FOUND.code) {
-        throw error;
-      }
-    };
 
-    try {
-      const serviceInfo = await getServiceInfo(lookupName, serviceName);
       if (serviceInfo.buf.length === 0) {
         const error = {};
         error.code = errConst.ERR_NO_SUCH_ENTRY.code;
         error.message = `Service not found. ${errConst.ERR_NO_SUCH_ENTRY.msg}`;
-        return reject(makeError(error.code, error.message));
+        throw makeError(error.code, error.message);
       }
+
       let serviceMd;
       try {
         serviceMd = await this.mutableData.fromSerial(serviceInfo.buf);
       } catch (e) {
         serviceMd = await this.mutableData.newPublic(serviceInfo.buf, consts.TAG_TYPE_WWW);
       }
-      const emulation = await serviceMd.emulateAs('NFS');
+
+      return serviceMd;
+    };
+
+    // Helper function to try different paths to find and
```

Comment:
@bochaco only thing I can think of here is maybe adding some tests for these helper functions. Otherwise all looking good

Reply:
Fully agreed, I think that should be the min benefit of splitting this function apart form readability. I'll work on it in the next few days.